iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/executor/RegionWriteExecutor.java

```diff
@@ -178,10 +179,11 @@ public RegionExecutionResult visitPlan(PlanNode node, WritePlanNodeExecutionCont
             status);
       } catch (ConsensusException e) {
         LOGGER.warn("Failed in the write API executing the consensus layer due to: ", e);
-        return RegionExecutionResult.create(
-            false,
-            e.toString(),
-            RpcUtils.getStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR, e.getMessage()));
+        TSStatus status = RpcUtils.getStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR, e.getMessage());
+        if (e instanceof ConsensusGroupNotExistException) {
+          status.setCode(TSStatusCode.WRITE_PROCESS_ERROR.getStatusCode());
+        }
+        return RegionExecutionResult.create(false, e.toString(), status);
```

Comment:
Looks fine to me. But I was thinking, since the migration + write has gone through multiple PRs, would it be better to use a more distinctive TSStatusCode uniformly for this type of issue? <img width="728" alt="image" src="https://github.com/user-attachments/assets/96c54664-9178-4d8f-87bf-4805e49f3cfb">

Reply:
Good Point! Changed.